core/src/main/java/org/neo4j/graphalgo/core/WeightMap.java

```diff
@@ -69,7 +57,7 @@ private double extractValue(Object value) {
 
     private void put(long key, double value) {
         if (weights == null) {
-            weights = new LongDoubleHashMap(capacity);
+            weights = new LongDoubleHashMap(capacity); // TODO move to ctor
```

Comment:
can we handle this todo here?
otherwise we will also potentially get NPEs in `get`